src/e2e/java/teammates/e2e/cases/e2e/InstructorSearchPageE2ETest.java

```diff
@@ -0,0 +1,121 @@
+package teammates.e2e.cases.e2e;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.InstructorCourseStudentDetailsEditPage;
+import teammates.e2e.pageobjects.InstructorCourseStudentDetailsViewPage;
+import teammates.e2e.pageobjects.InstructorSearchPage;
+import teammates.e2e.pageobjects.InstructorStudentRecordsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_SEARCH_PAGE}.
+ */
+public class InstructorSearchPageE2ETest extends BaseE2ETestCase {
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorSearchPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+        putDocuments(testData);
+    }
+
+    @Test
+    public void allTests() {
+
+        String instructorId = testData.accounts.get("instructor1OfCourse1").googleId;
+        AppUrl searchPageUrl = createUrl(Const.WebPageURIs.INSTRUCTOR_SEARCH_PAGE).withUserId(instructorId);
+
+        InstructorSearchPage searchPage = loginAdminToPage(searchPageUrl, InstructorSearchPage.class);
+
+        ______TS("cannot click search button if no checkbox is selected");
+
+        searchPage.clickStudentsCheckbox();
+        searchPage.verifyCannotClickSearchButton();
+
+        ______TS("search with no result");
+
+        searchPage.inputSearchContent("thiswillnothitanything");
+        searchPage.clickStudentsCheckbox();
+        searchPage.clickCommentsCheckbox();
+        searchPage.clickSearchButton();
```

Comment:
Maybe the 4 lines above can be abstracted as something like a `searchForStudentsAndComments` method in the page object to hide more of the element details in case maybe someday we stop using checkboxes or something.